core/src/main/scala/quasar/datasource/url/HttpAuthentication.scala

```diff
@@ -24,30 +24,32 @@ import cats.syntax.eq._
 
 import org.http4s._
 import org.http4s.headers.Authorization
+import org.http4s.util.{CaseInsensitiveString => CIS}
 
 import scodec.Codec
 import scodec.codecs._
 
 sealed trait HttpAuthentication extends Product with Serializable
 
 object HttpAuthentication {
-  final case class BearerToken(token: String) extends HttpAuthentication
+  final case class BearerToken(token: String, scheme: CIS) extends HttpAuthentication
 
   def apply[F[_]](authn: HttpAuthentication, req: Request[F]): Request[F] =
     authn match {
-      case BearerToken(tok) =>
-        req.putHeaders(Authorization(Credentials.Token(AuthScheme.Bearer, tok)))
+      case BearerToken(tok, scheme) =>
+        req.putHeaders(Authorization(Credentials.Token(scheme, tok)))
     }
 
-  def bearerToken(token: String): HttpAuthentication =
-    BearerToken(token)
+  def bearerToken(token: String, scheme: CIS): HttpAuthentication =
+    BearerToken(token, scheme)
 
   implicit val binaryCodec: Codec[HttpAuthentication] =
     discriminated[HttpAuthentication].by(uint8)
-      .| (0) { case BearerToken(t) => t } (BearerToken.apply) (utf8)
+      .| (0) { case BearerToken(t, s) => (t, s.value) } (tuple => BearerToken(tuple._1,CIS(tuple._2))) (utf8_32 ~ utf8)
```

Comment:
Here we'll want to preserve the decoding of the previous `0` case, inserting `"Bearer"` for compatibility, adding a new case for the updated format

```suggestion
      .caseO[String](0)(_ => None)(t => BearerToken(t, CIS("Bearer"))) (utf8)
      .caseP(1) { case BearerToken(t, s) => (t, s.value) } (tuple => BearerToken(tuple._1, CIS(tuple._2))) (utf8_32 ~ utf8)
```

Reply:
Smart!